cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+import com.wipro.ats.bdre.im.IMConstant;
+import com.wipro.ats.bdre.md.api.GetProperties;
+import com.wipro.ats.bdre.md.api.ProcessLog;
+import com.wipro.ats.bdre.md.beans.ProcessLogInfo;
+import org.apache.commons.cli.CommandLine;
+import org.apache.log4j.Logger;
+
+import java.sql.*;
+import java.util.*;
+import java.util.Date;
+
+/**
+ * Created by cloudera on 3/29/16.
+ */
+public class MigrationPreprocessor extends BaseStructure{
+    private static final Logger LOGGER = Logger.getLogger(MigrationPreprocessor.class);
+    private static String sourceRegularColumns=new String();   //comma-separated with datatype
+    private static String sourcePartitionColumns=new String(); //comma-separated with datatype
+    private static String stgPartitionColumns=new String();    //replacing source tech_partition with bdre tech partition
+    private static String stgTableDDL=new String();            //contains source stage table ddl
+    private static String destTableDDL=new String();           //contains destination table ddl
+    private static String filterCondition=new String();
+    private static String srcStgTableLocation=new String();
+    private static String destTableLocation=new String();
+    private static final String[][] PARAMS_STRUCTURE = {
+            {"pp", "parent-process-id", "Parent Process id of migration preprocessor"},
+            {"p", "process-id", " Process id of migration preprocessor"},
+            {"ied", "instance-exec-id", " instance exec id of preprocessor"}
+    };
+
+
+    protected static Connection getHiveJDBCConnection(String dbName, String hiveConnection) throws Exception {
+        try {
+            Class.forName(IMConstant.HIVE_DRIVER_NAME);
+            String hiveUser = IMConfig.getProperty("etl.hive-jdbcuser");
+            String hivePassword = IMConfig.getProperty("etl.hive-jdbcpassword");
+            Connection con = DriverManager.getConnection(hiveConnection + "/" + dbName, hiveUser, hivePassword);
+            con.createStatement().execute("set hive.exec.dynamic.partition.mode=nonstrict");
+            con.createStatement().execute("set hive.exec.dynamic.partition=true");
+            con.createStatement().execute("set hive.exec.max.dynamic.partitions.pernode=1000");
+            return con;
+        } catch (ClassNotFoundException e) {
+            throw new Exception(e);
+        } catch (SQLException e) {
+            throw new Exception(e);
+        }
+    }
+
+    public MigrationPreprocessorInfo execute(String[] params) throws Exception{
+
+        CommandLine commandLine = getCommandLine(params, PARAMS_STRUCTURE);
+        String parentProcessId=commandLine.getOptionValue("parent-process-id");
+        String processId = commandLine.getOptionValue("process-id");
+        String instanceExecId = commandLine.getOptionValue("instance-exec-id");
+        MigrationPreprocessorInfo mpInfo = prepareMigrate(parentProcessId,processId,instanceExecId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "mpInfo". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1488)